mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.swift

```diff
@@ -0,0 +1,141 @@
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                localKey = stringKey
+            }
+        }
+        return localKey
+    }
+
+    @objc func value(forCaseInsensitiveKey key: String) -> Any? {
+        var value : Any?
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                value = self[stringKey]
+            }
+        }
+        
+        return value
+    }
+
+    @objc func transformValuesToString() -> [String : Any] {
+        let transformedDictionary: [String : Any] = self.reduce(into: [:]) { result, element in
+            let key = element.key as! String
+            if let stringValue = element.value as? String {
+                result[key] = stringValue
+            } else if let stringValue = element.value as? NSString {
+                result[key] = stringValue
+            } else if let numberValue = element.value as? NSNumber {
+                result[key] = numberValue.stringValue
+            } else if let boolValue = element.value as? Bool {
+                result[key] = boolValue ? "true" : "false"
+            } else if let dateValue = element.value as? Date {
+                result[key] = MPDateFormatter.string(fromDateRFC1123: dateValue)
+            } else if let dateValue = element.value as? NSDate {
+                result[key] = MPDateFormatter.string(fromDateRFC1123: dateValue as Date)
+            } else if let dataValue = element.value as? Data, dataValue.count > 0 {
+                result[key] = String.init(data: dataValue, encoding: .utf8)
+            } else if let dataValue = element.value as? NSData, dataValue.length > 0 {
+                result[key] = String.init(data: dataValue as Data, encoding: .utf8)
+            } else if let dictValue = element.value as? [String : Any] {
+                result[key] = dictValue.description
+            } else if let dictValue = element.value as? NSDictionary {
+                result[key] = dictValue.description
+            } else if let dictValue = element.value as? NSMutableDictionary {
+                result[key] = dictValue.description
+            } else if let arrayValue = element.value as? [Any] {
+                result[key] = arrayValue.description
+            } else if let arrayValue = element.value as? NSArray {
+                result[key] = arrayValue.description
+            } else if let arrayValue = element.value as? NSMutableArray {
+                result[key] = arrayValue.description
+            } else if element.value is NSNull {
+                result[key] = "null"
+            } else {
+                MPLog.warning("Data type is not supported as an attribute value: \(type(of: element.value)) for key \(key)")
+            }
+        }
+        
+        return transformedDictionary
+    }
+}
+
+extension Dictionary {
```

Comment:
Once the tests are fixed and new tests added, this can probably be simplified so that you only write the implementations for NSDictionary or Swift dictionary and then cast to the other type. That way you don't need to duplicate the implementations. 

For sure it can be done for the `caseInsensitiveKey` and `value(forCaseInsensitiveKey key: String)` methods, but  `transformValuesToString` may have issues due to bridging messing up the types, though it would probably still be fine if the Swift implementation cases to NSDictionary vs the other way around.